src/Course/Alternative.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE RebindableSyntax #-}
+
+module Course.Alternative where
+
+import Course.Applicative
+import Course.Core
+import Course.Functor
+import Course.List
+import Course.Optional
+import Course.Parser
+import qualified Prelude as P(fmap, return, (>>=))
+
+-- | All instances of the `Alternative` type-class must satisfy three laws.
+-- These laws are not checked by the compiler. These laws are given as:
+--
+-- * The law of left identity
+--   `∀x. empty <|> x = x`
+--
+-- * The law of right identity
+--   `∀x. x <|> empty = x`
+--
+-- * The law of associativity
+--   `∀u v w. u <|> (v <|> w) = (u <|> v) <|> w`
+--
+-- You may notice that these are the same laws as Monoid.
```

Comment:
Good, but I think it needs more. Consider: "This is what the haddocks mean when they talk about \"a monoid on applicative functors\". Note that we are specifying the behavior on the type `k`, which is why we need a separate typeclass."

Reply:
I've had a go at this. I find it a bit difficult because I want to say "the Alternative should be related to the Applicative" but it's not clear to me how to make that more formal.
Perhaps I could just link to [this](https://wiki.haskell.org/Typeclassopedia#Laws_6) or [this](https://en.wikibooks.org/wiki/Haskell/Alternative_and_MonadPlus#Other_suggested_laws)

Input welcome